### PR TITLE
[FLINK-13436][end-to-end-tests] Add TPC-H queries as E2E tests

### PR DESCRIPTION
**Pull request by TsReaper**
## What is the purpose of the change

This PR adds TPC-H queries as end to end tests to flink.

## Brief change log

 - Add TPC-H queries as E2E tests

## Verifying this change

This change added tests and can be verified as follows: run the newly created `test_tpch.sh` test in end to end test module.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): yes
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? no

### CODE REVIEW

File: flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/config/ExecutionConfigOptions.java [hunk 1]
```diff
@@ -45,8 +45,8 @@
 	// ------------------------------------------------------------------------
 	public static final ConfigOption<Integer> TABLE_EXEC_SORT_DEFAULT_LIMIT =
 		key("table.exec.sort.default-limit")
-			.defaultValue(200)
-			.withDescription("Default limit when user don't set a limit after order by.");
+			.defaultValue(-1)
```
Comment (twalthr) [inline flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/config/ExecutionConfigOptions.java]: +1 for this change. I was also confused about a default limit which might confuse users.

File: flink-end-to-end-tests/flink-tpch-test/src/main/java/org/apache/flink/table/tpch/TpchResultComparator.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+							double a = Double.valueOf(actual[i]);
+							if (e < 0 && a > 0 || e > 0 && a < 0) {
+								failed = true;
+							} else {
+								if (e < 0) {
+									e = -e;
+									a = -a;
+								}
+								double t = round(a, 2);
+								// defined in TPCH standard specification v2.18.0 section 2.1.3.5
```
Comment (twalthr) [inline flink-end-to-end-tests/flink-tpch-test/src/main/java/org/apache/flink/table/tpch/TpchResultComparator.java]: nit: could you string replace `TPCH` for `TPC-H` in this PR for all comments and the name of the e2e test.

File: flink-end-to-end-tests/run-nightly-tests.sh [hunk 3]
```diff
@@ -152,6 +152,8 @@ run_test "SQL Client end-to-end test for Kafka 0.10" "$END_TO_END_DIR/test-scrip
 run_test "SQL Client end-to-end test for Kafka 0.11" "$END_TO_END_DIR/test-scripts/test_sql_client_kafka011.sh"
 run_test "SQL Client end-to-end test for modern Kafka" "$END_TO_END_DIR/test-scripts/test_sql_client_kafka.sh"
 
+run_test "TPCH end-to-end test (Blink planner)" "$END_TO_END_DIR/test-scripts/test_tpch.sh" "skip_check_exceptions"
```
Comment (twalthr) [inline flink-end-to-end-tests/run-nightly-tests.sh]: Are there exceptions that need to be skipped?

File: flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/config/ExecutionConfigOptions.java (hunk 1 above)
Comment (JingsongLi) [reply flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/config/ExecutionConfigOptions.java]: +1 for this change. The previous default value is just for stability.

**Comment by JingsongLi** [comment]
> Very nice PR @TsReaper. I just had minor comments. One question that I have is how do we ensure that the result verification is correct. Have you also manually tried to execute some queries in a popular system such as Oracle or Hive and compared it against Flink's results. I would just like to be on the safe side.

The library has its result: https://github.com/airlift/tpch/tree/master/src/main/resources/io/airlift/tpch/queries
Whatever the scale factor, the output is the same.
Now, we just compare with its result. I think its result is come from presto. And I agree with execute some queries in a popular system.

TPCH document also define some sample output for validation, (Like 2.4.1.4 in http://www.tpc.org/tpc_documents_current_versions/pdf/tpc-h_v2.18.0.pdf). But maybe it requires
 1 scale factor at least.

File: flink-end-to-end-tests/test-scripts/test-data/tpch/query/q1.sql [hunk 4]
```diff
@@ -0,0 +1,22 @@
+-- database: presto; groups: tpch; tables: lineitem
```
[the remark this answers is not in the document]
Comment (twalthr) [reply flink-end-to-end-tests/test-scripts/test-data/tpch/query/q1.sql]: What has been modified?

**Comment by TsReaper** [comment]
> Very nice PR @TsReaper. I just had minor comments. One question that I have is how do we ensure that the result verification is correct. Have you also manually tried to execute some queries in a popular system such as Oracle or Hive and compared it against Flink's results. I would just like to be on the safe side.

I've run the queries and the data on mysql and the results are compatible.

**Comment by TsReaper** [comment]
Oops. After modifying the default value of limit some plan validation tests failed. Let me fix it.

**Comment by KurtYoung** [comment]
@twalthr I and @wuchong have discussed the default limit today, and we both agree `-1` will be more appropriate default value.

**Comment by TsReaper** [comment]
Travis passed: https://travis-ci.com/TsReaper/flink/builds/121605193

**Pull request closed without merging**